autoload/vital/__vital__/VS/Vim/Window/FloatingWindow.vim

```diff
@@ -277,12 +310,36 @@ endif
 " exists
 "
 if has('nvim')
-  function! s:_exists(win) abort
-    return type(a:win) == type(0) && nvim_win_is_valid(a:win) && nvim_win_get_number(a:win) != -1
+  function! s:_exists(winid) abort
+    return type(a:winid) == type(0) && nvim_win_is_valid(a:winid) && nvim_win_get_number(a:winid) != -1
   endfunction
 else
-  function! s:_exists(win) abort
-    return type(a:win) == type(0) && winheight(a:win) != -1
+  function! s:_exists(winid) abort
+    return type(a:winid) == type(0) && winheight(a:winid) != -1
+  endfunction
+endif
+
+if has('nvim')
+  function! s:_info(winid) abort
+    let l:info = getwininfo(a:winid)[0]
+    return {
+    \   'row': l:info.winrow,
+    \   'col': l:info.wincol,
+    \   'width': l:info.width,
+    \   'height': l:info.height,
+    \   'topline': l:info.topline,
+    \ }
+  endfunction
+else
+  function! s:_info(winid) abort
+    let l:pos = popup_getpos(a:winid)
+    return {
+    \   'row': l:pos.core_line - 1,
+    \   'col': l:pos.core_col - 1,
+    \   'width': l:pos.core_width,
+    \   'height': l:pos.core_height,
+    \   'topline': l:pos.firstline,
+    \ }
   endfunction
```

Comment:
More detailed version of `s:Window.info` it will be useful re-calculate popup positions.